content/post/000-ipfs-in-web-browsers.md

```diff
@@ -0,0 +1,74 @@
+
+Initially we're aiming to ease access to the IPFS HTTP gateways - creating a bridge between the billions of people on the centralized web and content on the distributed web. We expect some experimentation in the next phase - adjusting node behavior based on browser APIs, device capabilities, bandwidth availability and power consumption - growing adoption with our browser partners as we learn, and building common understanding at standards bodies. Finally we aim to have multiple browsers implementing IPFS natively and interoperably.
+
+(INSERT STAGES DIAGRAM)
+
+## Firefox
+
+> Perhaps it makes sense to mention Firefox as a full recap-like section here, afaik we did not blog about libdweb at all yet. 
+> Idea: start with mentioning we've been working with Mozilla on initial APIs since 2018, mention [experimental protocol handler](https://github.com/ipfs-shipyard/ipfs-companion/pull/533) and [local discovery + TCP transport](https://github.com/ipfs-shipyard/ipfs-companion/pull/553) and include video demos from 2018: user-facing handler https://www.youtube.com/watch?v=fS8pLOQdOoM + optionally https://www.youtube.com/watch?v=FRzyWUXIyeo then end this section with hopes to see those APIs start landing in Nightly as a soft pressure by shining some light on this. 
+> Thoughts? [name=lidel]
+
+## Brave
+
+Brave has tended to live up to its name since the beginning - riling up the browser, publisher and crypto worlds. [Since 2018](https://github.com/brave/brave-browser/issues/819) we've been working with Brave on IPFS support in some form, and the results of that work are now emerging.
+
+Today you can open [brave://settings/extensions](brave://settings/extensions) in Brave beta or nightly and see a toggle for enabling IPFS Companion.
+
+![Enabling IPFS Companion in Brave](https://file.globalupload.io/ZVxjmPZ48E.gif)
+
+Removing this install step *really matters* for adoption. While technical users are familiar with browser extensions, it's less comfortable for non-technical users.
+
+In the future, when you [load a URL beginning with "ipfs://"](https://github.com/brave/brave-browser/issues/3045), Brave will prompt you to enable the extension, providing a smooth discovery and onboarding experience.
+
+But that isn't all. One of the biggest barriers to even just experimenting with true p2p networks in browsers is the lack of APIs available to extensions and web pages. The primitives of most p2p systems are filesystem access, raw TCP sockets, UDP sockets for broadcast and receive, and in browsers you need a way to navigate using a custom protocol. We've figured out one piece of that puzzle while working with Brave: Chromium OS sockets!
+
+[Chromium OS](https://en.wikipedia.org/wiki/Chromium_OS) has a raw sockets API, but it's disabled in the regular browser. Brave has modified their build to whitelist those APIs for IPFS Companion to use. 
```

Comment:
this makes the Brave extension more powerful than either the Firefox or Chrome extensions! (hint hint?)

Reply:
Fixed

Reply:
fixed